From 9600a7709045d796b48a49b5e6477d97a44849e2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Fri, 23 Feb 2024 00:45:32 +0200
Subject: [PATCH] [llvm-jitlink] [test] Add an XFAIL for a JITLink test on
 MinGW

This testcase fails on MinGW targets, because when compiling the
main() function, it gets an implicit call to __main(), which is
missing in this context.
---
 llvm/test/ExecutionEngine/JITLink/Generic/sectcreate.test | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/llvm/test/ExecutionEngine/JITLink/Generic/sectcreate.test b/llvm/test/ExecutionEngine/JITLink/Generic/sectcreate.test
index ec71011d545e..50585a2d02e9 100644
--- a/llvm/test/ExecutionEngine/JITLink/Generic/sectcreate.test
+++ b/llvm/test/ExecutionEngine/JITLink/Generic/sectcreate.test
@@ -8,4 +8,9 @@
 # Jitlink does not support ARM64 COFF files.
 # UNSUPPORTED: target=aarch64-pc-windows-{{.*}}
 
-# jitlink-check: *{4}foo = 0x2a2a5a5a
\ No newline at end of file
+# On MinGW targets, when compiling the main() function, it gets
+# an implicitly generated call to __main(), which is missing in
+# this context.
+# XFAIL: target={{.*}}-windows-gnu
+
+# jitlink-check: *{4}foo = 0x2a2a5a5a
-- 
2.25.1

